core/src/test/java/nl/nn/adapterframework/http/rest/ApiListenerPatternsTest.java

```diff
@@ -68,7 +68,9 @@ public ApiListenerPatternsTest(String pattern, String expected) {
 
 	@Test
 	public void testUriPattern() {
-		assertEquals(uriPattern, listener.getUriPattern());
+		String preFormattedUriPattern = uriPattern.startsWith("/") ? uriPattern : "/" + uriPattern;
+		String expectedUriPattern = preFormattedUriPattern.endsWith("/") ? preFormattedUriPattern.substring(0, preFormattedUriPattern.length()-1) : preFormattedUriPattern;
+		assertEquals(expectedUriPattern, listener.getUriPattern());
 	}
```

Comment:
Right now this is a 'hack' so you don't have to change the expected results. On line 39 to 56 you can see the input/output from the `setUriPattern(...)` and `getUriPattern()`
In this case, since the method returns a different value the test should reflect this, **(and not manipulate the test results)**.
I think, the `getUriPattern()` method as well as the `getCleanPattern()` should be tested, now that you moved some functionality over to other methods.